src/ir/datalog/program.h

```diff
@@ -158,6 +174,11 @@ class Rule {
   const Predicate& lhs() const { return lhs_; }
   const std::vector<Predicate>& rhs() const { return rhs_; }
 
+  bool operator==(const Rule& otherRule) const {
+    return this->lhs_ == otherRule.lhs() &&
+    this->rhs_ == otherRule.rhs();
```

Comment:
Use fields directly:
```suggestion
    return this->lhs_ == otherRule.lhs_ &&
    this->rhs_ == otherRule.rhs_;
```

Reply:
fixed